pkg/trigger/discovery_handler.go

```diff
@@ -185,10 +185,11 @@ func (mt *MigrationTrigger) processDiscoveryResource(r metav1.APIResource) {
 		utilruntime.HandleError(getErr)
 		return
 	}
-
-	stale := (getErr == nil && mt.staleStorageState(ss))
-	storageVersionChanged := (getErr == nil && ss.Status.CurrentStorageVersionHash != r.StorageVersionHash)
-	notFound := (getErr != nil && errors.IsNotFound(getErr))
+	found := getErr == nil || !errors.IsNotFound(getErr)
```

Comment:
Can you remove the "|| !errors.IsNotFound(getErr)"? That case is handled by the return above and it's weird to call this condition "found" if it contains error.

Reply:
@caesarxuchao Done